packages/components/src/FormField/FormFieldAffix.tsx

```diff
@@ -0,0 +1,68 @@
+import classnames from "classnames";
+import React, { RefObject } from "react";
+import { XOR } from "ts-xor";
+import { Affix, FormFieldProps, Suffix } from "./FormFieldTypes";
+import styles from "./FormField.css";
+import { Button } from "../Button";
+import { Icon } from "../Icon";
+
+interface AffixLabelProps extends Affix {
+  labelRef: RefObject<HTMLDivElement>;
+  variation?: "prefix" | "suffix";
+}
+
+export function AffixLabel({
+  label,
+  variation = "prefix",
+  labelRef,
+}: AffixLabelProps) {
+  const affixLabelClass = classnames(styles.affixLabel, {
+    [styles.suffix]: variation === "suffix",
+  });
+
+  return (
+    <div ref={labelRef} className={affixLabelClass}>
+      {label}
+    </div>
+  );
+}
+interface AffixIconProps extends Pick<FormFieldProps, "size"> {
+  readonly variation?: "prefix" | "suffix";
+}
+
+export function AffixIcon({
+  icon,
+  onClick,
+  ariaLabel,
+  variation = "prefix",
+  size,
+}: AffixIconProps & XOR<Affix, Suffix>) {
+  const affixIconClass = classnames(styles.affixIcon, {
+    [styles.suffix]: variation === "suffix",
+    [styles.hasAction]: onClick,
+  });
+
+  const iconSize = size === "small" ? "small" : "base";
```

Comment:
`size` is a prop that applies to the field. it can be a `small`, `base` or `large` field. However we never want to show a `large` icon. In this case we are only checking if the size is `small`.

Reply:
Hmm yeah, I guess that's fine. Was slightly worried about the `AffixIcon` component accepting a large size but internally it is actually ignoring it and always going with base. But, it's an internal component so if you're cool with it, i'm cool with it!